darts/models/filtering/kalman_filter.py

```diff
@@ -10,13 +10,86 @@
 import numpy as np
 import pandas as pd
 from nfoursid.kalman import Kalman
+from nfoursid.utils import Utils
 from nfoursid.nfoursid import NFourSID
 
 from darts.models.filtering.filtering_model import FilteringModel
 from darts.timeseries import TimeSeries
 from darts.logging import raise_if, raise_if_not
 
 
+def step_accepting_missing_obs(self, y: np.ndarray, u: np.ndarray):
+    """
+    Meant to be injected to a nfoursid.kalman.Kalman object.
+
+    Given an observed input ``u`` and output ``y``, update the filtered and predicted states of the Kalman filter.
+    Follows the implementation of the conventional Kalman filter in [1] on page 140.
+
+    This implementation also accepts missing outputs (``y`` set to `None`). In this case, the state of the
+    Kalman filter is updated, but without taking the measurement into account.
+
+    [1] Verhaegen, Michel, and Vincent Verdult. *Filtering and system identification: a least squares approach.*
+    Cambridge university press, 2007.
+    """
+    if y is not None:
+        Utils.validate_matrix_shape(y, (self.state_space.y_dim, 1), "y")
+    Utils.validate_matrix_shape(u, (self.state_space.u_dim, 1), "u")
+
+    x_pred = (
+        self.x_predicteds[-1]
+        if self.x_predicteds
+        else np.zeros((self.state_space.x_dim, 1))
+    )
+    p_pred = (
+        self.p_predicteds[-1] if self.p_predicteds else np.eye(self.state_space.x_dim)
+    )
+
+    if y is not None:
```

Comment:
Nice, I think this makes sense! It's equivalent to setting the gains to zero, right? Maybe doing that explicitly is clearer? This is consistent with https://github.com/pykalman/pykalman/blob/master/pykalman/standard.py#L289.

Reply:
Indeed I think this is equivalent, but the reason I didn't do it is computational:
* If we set K=0, we need the operand to multiply it with, which in turns requires `y` (and would crash if `y` is NaN)
* Even if I treated this above case, we would still need the rest of the operand, which requires computing some matrix multiplications for nothing.

Maybe I'll add a comment to the code to make this clear.